### PR TITLE
Use Model CRD for model build notification

### LINKED ISSUE #676: Replace gordo-notification-svc with a gordomodel CRD
Today we are posting a service every time a model is built. It probably makes more sense to notify about this using a custom resource.

### PR DESCRIPTION
**Pull request by milesgranger**
Will close #676

### CODE REVIEW

File: gordo_components/workflow/workflow_generator/resources/argo-workflow.yml.template [hunk 1]
```diff
@@ -788,15 +788,12 @@ spec:
                       - destination:
                           host: "{{'{{inputs.parameters.host-name}}'}}"
 
-  # SVC to notify watchman that a model has been built and can be served by the ML Server
-  # This should only be temporary, as it's not _really_ a service, but used to notify Watchman
-  # that the model is built and should be able to be contacted via the main server.
-  # Here until we create a 'GordoModel' CRD which can then notify that the model is built.
-  - name: gordo-notification-svc
+  # CRD to notify watchman that a model has been built and can be served by the ML Server
+  - name: gordo-model-crd
```
Comment (epa095) [inline gordo_components/workflow/workflow_generator/resources/argo-workflow.yml.template]: not really a `crd`

**Review by epa095 — COMMENTED** [review]
looking gooder and goodier

**Comment by milesgranger** [comment]
Ran it through the test-project on gordotest46 'test-milesg1' :heavy_check_mark:

**Review by epa095 — APPROVED** [review]
Baffeling

**Pull request merged**